app/src/main/java/worker/file/DownloadFile.java

```diff
@@ -0,0 +1,69 @@
+package worker.file;
+
+import capyfile.rmi.DownloadFileArgs;
+import capyfile.rmi.DownloadFileRes;
+import com.healthmarketscience.rmiio.GZIPRemoteInputStream;
+import com.healthmarketscience.rmiio.RemoteInputStreamServer;
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.security.InvalidParameterException;
+import java.util.stream.IntStream;
+import worker.config.Config;
+
+public class DownloadFile
+{
+	public static DownloadFileRes downloadFile (DownloadFileArgs args) throws Exception
+	{
+		// volume exists
+
+		if (!IntStream.of (Config.getAvailableVolumes ()).anyMatch (x -> x == args.volume)) {
+			throw new InvalidParameterException ();
+		}
+
+		RemoteInputStreamServer istream = null;
+
+		try {
+
+			File file = new File (String.format (
+				"%s/files/volume%d/%s", Config.getVolumeBasePath (), args.volume,
+				args.uuid.toString ()));
+
+			// fallback to backup
+			// NOTE: This could have problems in NFS
```

Comment:
Why? Do you mean the fallback to the backup folder or the bytes streaming?

Reply:
I mean the function used to check whether or not a file exists. I did read some comment in a thread from about 2012 where a guy talked about it. However if it's a bug with the IO API it probably should have been fixed by now. So for now it's just a warning if anything happens.

Reply:
Ok, thanks for warning us. Consider sharing us the thread if you find it again.